hack/deployer/runner/ocp.go

```diff
@@ -186,7 +167,54 @@ func (d *OcpDriver) Execute() error {
 	return err
 }
 
-func (d *OcpDriver) clusterExists() (bool, error) {
+func (d *OcpDriver) ensureContext() (cleanUp func(), err error) {
+	cleanUp = func() {} // NOOP
+	if d.ctx["WorkDir"].(string) == "" {
+		dir, err := ioutil.TempDir("", d.ctx["ClusterName"].(string))
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		cleanUp = func() {
+			os.RemoveAll(dir)
+		}
+		d.ctx["WorkDir"] = dir
+		log.Printf("Created WorkDir: %s", d.ctx["WorkDir"])
+	}
+
+	if _, exists := d.ctx["ClusterStateDir"]; !exists {
+		stateDir := filepath.Join(d.ctx["WorkDir"].(string), d.ctx["ClusterName"].(string))
+		if err := os.MkdirAll(stateDir, os.ModePerm); err != nil {
+			return cleanUp, err
```

Comment:
Is it possible that `cleanup` could be a noop because `d.ctx["WorkDir"]` was not empty? Is that the expected behaviour?

Reply:
Yes.